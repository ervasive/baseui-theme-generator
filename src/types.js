// @flow
export { ColorsT } from "baseui/styles/types.js.flow";

export type VariantT = {
  color: string,
  contrast: ?number,
  hueOffset: ?number,
  "50": ?string,
  "100": ?string,
  "200": ?string,
  "300": ?string,
  "400": ?string,
  "500": ?string,
  "600": ?string,
  "700": ?string,
  "800": ?string,
  "900": ?string,
  "1000": ?string
};

export type PaletteT = {
  primary: VariantT,
  negative: VariantT,
  warning: VariantT,
  positive: VariantT,
  mono: VariantT
};

export type NativePaletteColorsT = {
  // Primary Palette
  primary: string,
  primary50: string,
  primary100: string,
  primary200: string,
  primary300: string,
  primary400: string,
  primary500: string,
  primary600: string,
  primary700: string,
  primary800: string,
  primary900: string,
  primary1000: string,

  // Negative Palette
  negative: string,
  negative50: string,
  negative100: string,
  negative200: string,
  negative300: string,
  negative400: string,
  negative500: string,
  negative600: string,
  negative700: string,
  negative800: string,
  negative900: string,
  negative1000: string,

  // Warning Palette
  warning: string,
  warning50: string,
  warning100: string,
  warning200: string,
  warning300: string,
  warning400: string,
  warning500: string,
  warning600: string,
  warning700: string,
  warning800: string,
  warning900: string,
  warning1000: string,

  // Positive Palette
  positive: string,
  positive50: string,
  positive100: string,
  positive200: string,
  positive300: string,
  positive400: string,
  positive500: string,
  positive600: string,
  positive700: string,
  positive800: string,
  positive900: string,
  positive1000: string,

  // Mono Palette
  mono: string,
  mono50: string,
  mono100: string,
  mono200: string,
  mono300: string,
  mono400: string,
  mono500: string,
  mono600: string,
  mono700: string,
  mono800: string,
  mono900: string,
  mono1000: string,
  white: string,
  black: string
};
