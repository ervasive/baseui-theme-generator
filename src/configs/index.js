export {default as lightThemeConfig} from './light'
export {default as darkThemeConfig} from './dark'
