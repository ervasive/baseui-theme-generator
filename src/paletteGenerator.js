// @flow

import { lighten, adjustHue } from "polished";

import {
  type PaletteMapT,
  type NativePaletteColorsT,
  type VariantT
} from "./types";

export default function generate(palette: PaletteMapT): NativePaletteColorsT {
  let colors = {};

  for (let [colorName, color] of palette.entries()) {
    const generatedVariants = generateVariants(color);

    delete color.contrast;
    delete color.hueOffset;

    const variants = { ...generatedVariants, ...color };

    for (let [variantName, variantValue] of Object.entries(variants)) {
      colors[`${colorName}${variantName}`] = variantValue;

      if (colorName === "mono") {
        colors.white = "#fff";
        colors.black = "#000";
      } else {
        colors[colorName] = variants[400];
      }
    }
  }

  return colors;
}

export function generateVariants(variant: VariantT): { [string]: string } {
  const colors = {};
  let color, hueOffset, contrast;

  if (typeof variant === "string") {
    color = variant;
    contrast = 0.1;
    hueOffset = 0;
  } else {
    color = variant[400];
    contrast = variant.contrast ? variant.contrast : 0.1;
    hueOffset = variant.hueOffset ? variant.hueOffset : 0;
  }

  new Array(11).fill(color).map((c, i, a) => {
    const index = i === 0 ? 50 : i * 100;
    const multiplier = a.length / 2 - i;
    const h = adjustHue(hueOffset * multiplier + 360, c);
    colors[index] = lighten(contrast * multiplier, String(h));
  });

  return colors;
}
