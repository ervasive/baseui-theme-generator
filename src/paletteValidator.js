// @flow
import { parseToRgb } from "polished";

import { type PaletteT } from "./types";

const colorShape = [
  `{`,
  `\tcolor:      string (css color value), required`,
  `\tcontrast?:  number (from 0.0 to 1.0)`,
  `\thueOffset?: number (degrees)`,
  `\t50?:        string (css color value)`,
  `\t100?:       string (css color value)`,
  `\t200?:       string (css color value)`,
  `\t300?:       string (css color value)`,
  `\t400?:       string (css color value) equivalent to 'color' property`,
  `\t500?:       string (css color value)`,
  `\t600?:       string (css color value)`,
  `\t700?:       string (css color value)`,
  `\t800?:       string (css color value)`,
  `\t900?:       string (css color value)`,
  `\t1000?:      string (css color value)`,
  `}`
].join(`\n`);

export default function validate(palette: PaletteT): void {
  if (
    !palette ||
    typeof palette !== "object" ||
    !(
      palette.hasOwnProperty("primary") &&
      palette.hasOwnProperty("negative") &&
      palette.hasOwnProperty("warning") &&
      palette.hasOwnProperty("positive") &&
      palette.hasOwnProperty("mono")
    )
  ) {
    throw new Error(
      `Invalid palette.\n Make sure that all required color objects are ` +
        `provided (primary, negative, warning, positive, mono) and have the ` +
        `following shape: \n${colorShape}`
    );
  }

  for (let [colorName, color] of Object.entries(palette)) {
    if (
      typeof color !== "object" ||
      color === null ||
      !color.hasOwnProperty("color")
    ) {
      throw new Error(
        `Invalid color.\n The color: '${colorName}' misses the required ` +
          `'color' property. Make sure that all provided color objects have ` +
          `the following shape: \n${colorShape}`
      );
    }

    [
      "color",
      "50",
      "100",
      "200",
      "300",
      "400",
      "500",
      "600",
      "700",
      "800",
      "900",
      "1000"
    ].map(c => {
      try {
        color.hasOwnProperty(c) && parseToRgb(String(color[c]));
      } catch (e) {
        throw new Error(
          `Unable to validate '${colorName}[${c}]' color variant.\n` +
            `The '${String(color[c])}' is not a valid CSS color. ` +
            `Make sure that all provided color objects have ` +
            `the following shape: \n${colorShape}`
        );
      }
    });

    if (
      color.hasOwnProperty("contrast") &&
      (Number(color.contrast) > 1 || Number(color.contrast) < 0)
    ) {
      throw new Error(
        `The contrast value of '${colorName}' color is out of allowed range.\n` +
          `Make sure that all provided color objects have ` +
          `the following shape: \n${colorShape}`
      );
    }

    if (
      color.hasOwnProperty("hueOffset") &&
      isNaN(parseInt(color.hueOffset, 10))
    ) {
      throw new Error(
        `Unable to convert the 'hueOffset' value of '${colorName}' color to a ` +
          `number.\n` +
          `Make sure that all provided color objects have ` +
          `the following shape: \n${colorShape}`
      );
    }
  }
}
