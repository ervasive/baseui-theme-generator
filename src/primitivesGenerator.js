// @flow
import merge from 'lodash/merge'
import {lighten, adjustHue} from 'polished'
import {validateThemeConfig} from './themeConfigValidator'
import {lightThemeConfig, darkThemeConfig} from './configs'

import {
  type ThemeConfigT,
  type ThemeConfigColorT,
  type ThemePrimitivesT,
} from './types'

export function generatePrimitives(config?: ThemeConfigT): ThemePrimitivesT {
  validateThemeConfig(config)

  const completeConfig = merge(
    {},
    config && config.type === 'dark' ? darkThemeConfig : lightThemeConfig,
    config,
  )

  const primitives = {}
  primitives.type = completeConfig.type

  for (let colorName of Object.keys(completeConfig.palette)) {
    const color =
      config && config.palette && config.palette[colorName]
        ? config.palette[colorName]
        : completeConfig.palette[colorName]

    let colorVariants = generateVariants(color)

    if (typeof color === 'object') {
      colorVariants = {...colorVariants, ...color}
    }

    for (let [variantName, variantValue] of Object.entries(colorVariants)) {
      primitives[`${colorName}${variantName}`] = variantValue

      if (colorName === 'mono') {
        primitives.white = '#fff'
        primitives.black = '#000'
      } else {
        primitives[colorName] = colorVariants['400']
      }
    }
  }

  primitives.primaryFontFamily = completeConfig.typography.primaryFontFamily

  return primitives
}

export function generateVariants(color: ThemeConfigColorT): {[string]: string} {
  const colors = {}
  let baseColor, hueOffset, contrast

  if (typeof color === 'string') {
    baseColor = color
    contrast = 0.1
    hueOffset = 0
  } else {
    baseColor = color['400']
    contrast = color.contrast || 0.1
    hueOffset = color.hueOffset || 0
  }

  ;[
    {index: 50, multiplier: 4},
    {index: 100, multiplier: 3},
    {index: 200, multiplier: 2},
    {index: 300, multiplier: 1},
    {index: 400, multiplier: 0},
    {index: 500, multiplier: -1},
    {index: 600, multiplier: -2},
    {index: 700, multiplier: -3},
    {index: 800, multiplier: -4},
    {index: 900, multiplier: -5},
    {index: 1000, multiplier: -6},
  ].map(({index, multiplier}) => {
    if (color[index]) {
      colors[index] = color[index]
    } else {
      const h = adjustHue(hueOffset * multiplier + 360, baseColor)
      colors[index] = lighten(contrast * multiplier, String(h))
    }
  })

  return colors
}
