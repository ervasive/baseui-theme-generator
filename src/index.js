// @flow
import deepMerge from 'baseui/utils/deep-merge'
import {validateThemeConfig} from './themeConfigValidator'
import {generatePrimitives} from './primitivesGenerator'
import {generateColors} from './colorsGenerator'

import {type ThemeConfigT} from './types'

export function generateTheme(
  theme: ThemeConfigT,
  overrides?: Function = () => {},
) {
  try {
    validateThemeConfig(theme)

    const themePrimitives = generatePrimitives(theme)
    const {tokens, themed} = generateColors(themePrimitives, theme.palette.type)

    let primaryFontFamily

    if (!theme.typography || !theme.typography.primaryFontFamily) {
      primaryFontFamily =
        'system-ui, "Helvetica Neue", Helvetica, Arial, sans-serif'
    } else {
      primaryFontFamily = theme.typography.primaryFontFamily
    }

    const result = {
      colors: tokens,
      breakpoints: {
        small: 320,
        medium: 600,
        large: 1280,
      },
      typography: {
        font100: {
          fontFamily: primaryFontFamily,
          fontSize: '11px',
          fontWeight: 'normal',
          lineHeight: '16px',
        },
        font200: {
          fontFamily: primaryFontFamily,
          fontSize: '12px',
          fontWeight: 'normal',
          lineHeight: '20px',
        },
        font250: {
          fontFamily: primaryFontFamily,
          fontSize: '12px',
          fontWeight: 'bold',
          lineHeight: '20px',
        },
        font300: {
          fontFamily: primaryFontFamily,
          fontSize: '14px',
          fontWeight: 'normal',
          lineHeight: '20px',
        },
        font350: {
          fontFamily: primaryFontFamily,
          fontSize: '14px',
          fontWeight: 'bold',
          lineHeight: '20px',
        },
        font400: {
          fontFamily: primaryFontFamily,
          fontSize: '16px',
          fontWeight: 'normal',
          lineHeight: '24px',
        },
        font450: {
          fontFamily: primaryFontFamily,
          fontSize: '16px',
          fontWeight: 'bold',
          lineHeight: '24px',
        },
        font500: {
          fontFamily: primaryFontFamily,
          fontSize: '20px',
          fontWeight: 'bold',
          lineHeight: '28px',
        },
        font600: {
          fontFamily: primaryFontFamily,
          fontSize: '24px',
          fontWeight: 'bold',
          lineHeight: '36px',
        },
        font700: {
          fontFamily: primaryFontFamily,
          fontSize: '32px',
          fontWeight: 'bold',
          lineHeight: '48px',
        },
        font800: {
          fontFamily: primaryFontFamily,
          fontSize: '40px',
          fontWeight: 'bold',
          lineHeight: '56px',
        },
        font900: {
          fontFamily: primaryFontFamily,
          fontSize: '52px',
          fontWeight: 'bold',
          lineHeight: '68px',
        },
        font1000: {
          fontFamily: primaryFontFamily,
          fontSize: '72px',
          fontWeight: 'normal',
          lineHeight: '96px',
        },
        font1100: {
          fontFamily: primaryFontFamily,
          fontSize: '96px',
          fontWeight: 'normal',
          lineHeight: '116px',
        },
      },
      sizing: {
        scale0: '2px',
        scale100: '4px',
        scale200: '6px',
        scale300: '8px',
        scale400: '10px',
        scale500: '12px',
        scale550: '14px',
        scale600: '16px',
        scale700: '20px',
        scale800: '24px',
        scale900: '32px',
        scale1000: '40px',
        scale1200: '48px',
        scale1400: '56px',
        scale1600: '64px',
        scale2400: '96px',
        scale3200: '128px',
        scale4800: '192px',
      },
      lighting: {
        shadow400: themed(
          `0 1px 4px hsla(0, 0%, 0%, 0.16)`,
          `0 1px 4px hsla(0, 0%, 0%, 0.16)`,
        ),
        shadow500: themed(
          `0 2px 8px hsla(0, 0%, 0%, 0.16)`,
          `0 1px 4px hsla(0, 0%, 0%, 0.16)`,
        ),
        shadow600: themed(
          `0 4px 16px hsla(0, 0%, 0%, 0.16)`,
          `0 4px 16px hsla(0, 0%, 0%, 0.16)`,
        ),
        shadow700: themed(
          `0 8px 24px hsla(0, 0%, 0%, 0.16)`,
          `0 8px 24px hsla(0, 0%, 0%, 0.16)`,
        ),
        overlay0: themed(
          `inset 0 0 0 1000px hsla(0, 0%, 0%, 0)`,
          `inset 0 0 0 1000px hsla(0, 0%, 0%, 0)`,
        ),
        overlay100: themed(
          `inset 0 0 0 1000px hsla(0, 0%, 0%, 0.04)`,
          `inset 0 0 0 1000px hsla(0, 0%, 0%, 0.04)`,
        ),
        overlay200: themed(
          `inset 0 0 0 1000px hsla(0, 0%, 0%, 0.08)`,
          `inset 0 0 0 1000px hsla(0, 0%, 0%, 0.08)`,
        ),
        overlay300: themed(
          `inset 0 0 0 1000px hsla(0, 0%, 0%, 0.12)`,
          `inset 0 0 0 1000px hsla(0, 0%, 0%, 0.12)`,
        ),
        overlay400: themed(
          `inset 0 0 0 1000px hsla(0, 0%, 0%, 0.16)`,
          `inset 0 0 0 1000px hsla(0, 0%, 0%, 0.16)`,
        ),
        overlay500: themed(
          `inset 0 0 0 1000px hsla(0, 0%, 0%, 0.16)`,
          `inset 0 0 0 1000px hsla(0, 0%, 0%, 0.16)`,
        ),
        overlay600: themed(
          `inset 0 0 0 1000px hsla(0, 0%, 0%, 0.24)`,
          `inset 0 0 0 1000px hsla(0, 0%, 0%, 0.24)`,
        ),
      },
      borders: {
        border100: {
          borderColor: themed(
            'hsla(0, 0%, 0%, 0.04)',
            'hsla(0, 0%, 100%, 0.02)',
          ),
          borderStyle: 'solid',
          borderWidth: '1px',
        },
        border200: {
          borderColor: themed(
            'hsla(0, 0%, 0%, 0.08)',
            'hsla(0, 0%, 100%, 0.05)',
          ),
          borderStyle: 'solid',
          borderWidth: '1px',
        },
        border300: {
          borderColor: themed(
            'hsla(0, 0%, 0%, 0.12)',
            'hsla(0, 0%, 100%, 0.08)',
          ),
          borderStyle: 'solid',
          borderWidth: '1px',
        },
        border400: {
          borderColor: themed(
            'hsla(0, 0%, 0%, 0.16)',
            'hsla(0, 0%, 100%, 0.1)',
          ),
          borderStyle: 'solid',
          borderWidth: '1px',
        },
        border500: {
          borderColor: themed(
            'hsla(0, 0%, 0%, 0.2)',
            'hsla(0, 0%, 100%, 0.12)',
          ),
          borderStyle: 'solid',
          borderWidth: '1px',
        },
        border600: {
          borderColor: themed(
            'hsla(0, 0%, 0%, 0.24)',
            'hsla(0, 0%, 100%, 0.15)',
          ),
          borderStyle: 'solid',
          borderWidth: '1px',
        },
        radius100: '2px',
        radius200: '4px',
        radius300: '8px',
        radius400: '12px',
        useRoundedCorners: true,
      },
      animation: {
        timing100: '0.25s',
        timing400: '0.4s',
        timing700: '0.6s',
        easeOutCurve: 'cubic-bezier(.2, .8, .4, 1)',
        easeInCurve: 'cubic-bezier(.8, .2, .6, 1)',
        easeInOutCurve: 'cubic-bezier(0.4, 0, 0.2, 1)',
      },
      zIndex: {
        modal: 2000,
      },
      tooltip: {
        backgroundColor: themed('mono900', 'mono200'),
      },
    }

    return deepMerge(result, overrides(result, themed))
  } catch (e) {
    // eslint-disable-next-line
    console.log(e.message)
  }
}
