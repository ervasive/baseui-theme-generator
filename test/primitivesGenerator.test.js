import {generatePrimitives, generateVariants} from '../src/primitivesGenerator'

describe('Generate primitives', () => {
  it('should generate default light primitives if no or empty config was provided', () => {
    expect(generatePrimitives()).toMatchSnapshot()
    expect(generatePrimitives({})).toMatchSnapshot()
  })

  it(`should generate a primitives object based on custom theme config`, () => {
    expect(generatePrimitives({palette: {primary: 'red'}})).toMatchSnapshot()

    expect(
      generatePrimitives({palette: {primary: {400: 'red'}}}),
    ).toMatchSnapshot()
  })

  it(`should preserve explicitly defined color varians in a generated primitives object`, () => {
    expect(
      generatePrimitives({palette: {primary: {400: 'red', 500: 'green'}}}),
    ).toMatchSnapshot()
  })
})

describe('Color variants generator', () => {
  it('should generate correct color modulations with default params', () => {
    expect(generateVariants({400: '#ff0000'})).toMatchSnapshot()
  })

  it('should generate correct color modulations with custom contrast value', () => {
    expect(generateVariants({400: '#ff0000', contrast: 0.15})).toMatchSnapshot()
  })

  it('should generate correct color modulations with custom hueOffset value', () => {
    expect(generateVariants({400: '#ff0000', hueOffset: 15})).toMatchSnapshot()
  })
  it('should generate correct color modulations with custom contrast and hueOffset values', () => {
    expect(
      generateVariants({
        400: '#ff0000',
        contrast: 0.15,
        hueOffset: 15,
      }),
    ).toMatchSnapshot()
  })
})
