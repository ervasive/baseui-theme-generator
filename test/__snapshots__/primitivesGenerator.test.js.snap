// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color variants generator should generate correct color modulations with custom contrast and hueOffset values 1`] = `
Object {
  "100": "#fff9e5",
  "1000": "#000",
  "200": "#fc9",
  "300": "#ff794d",
  "400": "#ff0000",
  "50": "#fff",
  "500": "#b3002d",
  "600": "#603",
  "700": "#1a0013",
  "800": "#000",
  "900": "#000",
}
`;

exports[`Color variants generator should generate correct color modulations with custom contrast value 1`] = `
Object {
  "100": "#ffe5e5",
  "1000": "#000",
  "200": "#f99",
  "300": "#ff4d4d",
  "400": "#ff0000",
  "50": "#fff",
  "500": "#b30000",
  "600": "#600",
  "700": "#1a0000",
  "800": "#000",
  "900": "#000",
}
`;

exports[`Color variants generator should generate correct color modulations with custom hueOffset value 1`] = `
Object {
  "100": "#ffe599",
  "1000": "#000",
  "200": "#ffb366",
  "300": "#f63",
  "400": "#ff0000",
  "50": "#ffc",
  "500": "#c03",
  "600": "#99004d",
  "700": "#66004c",
  "800": "#303",
  "900": "#000",
}
`;

exports[`Color variants generator should generate correct color modulations with default params 1`] = `
Object {
  "100": "#f99",
  "1000": "#000",
  "200": "#f66",
  "300": "#f33",
  "400": "#ff0000",
  "50": "#fcc",
  "500": "#c00",
  "600": "#900",
  "700": "#600",
  "800": "#300",
  "900": "#000",
}
`;

exports[`Generate primitives should generate a primitives object based on custom theme config 1`] = `
Object {
  "black": "#000",
  "mono100": "#FFFFFF",
  "mono1000": "#000000",
  "mono200": "#F7F7F7",
  "mono300": "#F0F0F0",
  "mono400": "#E5E5E5",
  "mono50": "#fff",
  "mono500": "#CCCCCC",
  "mono600": "#B3B3B3",
  "mono700": "#999999",
  "mono800": "#666666",
  "mono900": "#333333",
  "negative": "#E54937",
  "negative100": "#FADBD7",
  "negative1000": "#000",
  "negative200": "#F4AFA7",
  "negative300": "#EB7567",
  "negative400": "#E54937",
  "negative50": "#FDF0EF",
  "negative500": "#AE372A",
  "negative600": "#892C21",
  "negative700": "#5C1D16",
  "negative800": "#471009",
  "negative900": "#1a0603",
  "positive": "#07A35A",
  "positive100": "#CDEDDE",
  "positive1000": "#000",
  "positive200": "#88D3B0",
  "positive300": "#43B982",
  "positive400": "#07A35A",
  "positive50": "#EBF8F2",
  "positive500": "#057C44",
  "positive600": "#046236",
  "positive700": "#034124",
  "positive800": "#000",
  "positive900": "#000",
  "primary": "#f00",
  "primary100": "#f99",
  "primary1000": "#000",
  "primary200": "#f66",
  "primary300": "#f33",
  "primary400": "#f00",
  "primary50": "#fcc",
  "primary500": "#c00",
  "primary600": "#900",
  "primary700": "#600",
  "primary800": "#300",
  "primary900": "#000",
  "primaryFontFamily": "system-ui, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
  "rating": "#FFC043",
  "rating100": "#fff3dc",
  "rating1000": "#100b00",
  "rating200": "#FFE1A5",
  "rating300": "#ffd176",
  "rating400": "#FFC043",
  "rating50": "#fff",
  "rating500": "#ffaf10",
  "rating600": "#dc9200",
  "rating700": "#a97000",
  "rating800": "#764e00",
  "rating900": "#432d00",
  "type": "light",
  "warning": "#ED6F0E",
  "warning100": "#FBE2CF",
  "warning1000": "#000",
  "warning200": "#F6BA8B",
  "warning300": "#F19248",
  "warning400": "#ED6F0E",
  "warning50": "#FEF3EC",
  "warning500": "#B4540B",
  "warning600": "#8E4308",
  "warning700": "#5F2C06",
  "warning800": "#2c1503",
  "warning900": "#000",
  "white": "#fff",
}
`;

exports[`Generate primitives should generate a primitives object based on custom theme config 2`] = `
Object {
  "black": "#000",
  "mono100": "#FFFFFF",
  "mono1000": "#000000",
  "mono200": "#F7F7F7",
  "mono300": "#F0F0F0",
  "mono400": "#E5E5E5",
  "mono50": "#fff",
  "mono500": "#CCCCCC",
  "mono600": "#B3B3B3",
  "mono700": "#999999",
  "mono800": "#666666",
  "mono900": "#333333",
  "negative": "#E54937",
  "negative100": "#FADBD7",
  "negative1000": "#000",
  "negative200": "#F4AFA7",
  "negative300": "#EB7567",
  "negative400": "#E54937",
  "negative50": "#FDF0EF",
  "negative500": "#AE372A",
  "negative600": "#892C21",
  "negative700": "#5C1D16",
  "negative800": "#471009",
  "negative900": "#1a0603",
  "positive": "#07A35A",
  "positive100": "#CDEDDE",
  "positive1000": "#000",
  "positive200": "#88D3B0",
  "positive300": "#43B982",
  "positive400": "#07A35A",
  "positive50": "#EBF8F2",
  "positive500": "#057C44",
  "positive600": "#046236",
  "positive700": "#034124",
  "positive800": "#000",
  "positive900": "#000",
  "primary": "red",
  "primary100": "#f99",
  "primary1000": "#000",
  "primary200": "#f66",
  "primary300": "#f33",
  "primary400": "red",
  "primary50": "#fcc",
  "primary500": "#c00",
  "primary600": "#900",
  "primary700": "#600",
  "primary800": "#300",
  "primary900": "#000",
  "primaryFontFamily": "system-ui, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
  "rating": "#FFC043",
  "rating100": "#fff3dc",
  "rating1000": "#100b00",
  "rating200": "#FFE1A5",
  "rating300": "#ffd176",
  "rating400": "#FFC043",
  "rating50": "#fff",
  "rating500": "#ffaf10",
  "rating600": "#dc9200",
  "rating700": "#a97000",
  "rating800": "#764e00",
  "rating900": "#432d00",
  "type": "light",
  "warning": "#ED6F0E",
  "warning100": "#FBE2CF",
  "warning1000": "#000",
  "warning200": "#F6BA8B",
  "warning300": "#F19248",
  "warning400": "#ED6F0E",
  "warning50": "#FEF3EC",
  "warning500": "#B4540B",
  "warning600": "#8E4308",
  "warning700": "#5F2C06",
  "warning800": "#2c1503",
  "warning900": "#000",
  "white": "#fff",
}
`;

exports[`Generate primitives should generate default light primitives if no or empty config was provided 1`] = `
Object {
  "black": "#000",
  "mono100": "#FFFFFF",
  "mono1000": "#000000",
  "mono200": "#F7F7F7",
  "mono300": "#F0F0F0",
  "mono400": "#E5E5E5",
  "mono50": "#fff",
  "mono500": "#CCCCCC",
  "mono600": "#B3B3B3",
  "mono700": "#999999",
  "mono800": "#666666",
  "mono900": "#333333",
  "negative": "#E54937",
  "negative100": "#FADBD7",
  "negative1000": "#000",
  "negative200": "#F4AFA7",
  "negative300": "#EB7567",
  "negative400": "#E54937",
  "negative50": "#FDF0EF",
  "negative500": "#AE372A",
  "negative600": "#892C21",
  "negative700": "#5C1D16",
  "negative800": "#471009",
  "negative900": "#1a0603",
  "positive": "#07A35A",
  "positive100": "#CDEDDE",
  "positive1000": "#000",
  "positive200": "#88D3B0",
  "positive300": "#43B982",
  "positive400": "#07A35A",
  "positive50": "#EBF8F2",
  "positive500": "#057C44",
  "positive600": "#046236",
  "positive700": "#034124",
  "positive800": "#000",
  "positive900": "#000",
  "primary": "#276EF1",
  "primary100": "#D2E0FC",
  "primary1000": "#000",
  "primary200": "#9CBCF8",
  "primary300": "#548BF4",
  "primary400": "#276EF1",
  "primary50": "#EDF3FE",
  "primary500": "#174EB6",
  "primary600": "#123D90",
  "primary700": "#0C2960",
  "primary800": "#051c47",
  "primary900": "#020917",
  "primaryFontFamily": "system-ui, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
  "rating": "#FFC043",
  "rating100": "#fff3dc",
  "rating1000": "#100b00",
  "rating200": "#FFE1A5",
  "rating300": "#ffd176",
  "rating400": "#FFC043",
  "rating50": "#fff",
  "rating500": "#ffaf10",
  "rating600": "#dc9200",
  "rating700": "#a97000",
  "rating800": "#764e00",
  "rating900": "#432d00",
  "type": "light",
  "warning": "#ED6F0E",
  "warning100": "#FBE2CF",
  "warning1000": "#000",
  "warning200": "#F6BA8B",
  "warning300": "#F19248",
  "warning400": "#ED6F0E",
  "warning50": "#FEF3EC",
  "warning500": "#B4540B",
  "warning600": "#8E4308",
  "warning700": "#5F2C06",
  "warning800": "#2c1503",
  "warning900": "#000",
  "white": "#fff",
}
`;

exports[`Generate primitives should generate default light primitives if no or empty config was provided 2`] = `
Object {
  "black": "#000",
  "mono100": "#FFFFFF",
  "mono1000": "#000000",
  "mono200": "#F7F7F7",
  "mono300": "#F0F0F0",
  "mono400": "#E5E5E5",
  "mono50": "#fff",
  "mono500": "#CCCCCC",
  "mono600": "#B3B3B3",
  "mono700": "#999999",
  "mono800": "#666666",
  "mono900": "#333333",
  "negative": "#E54937",
  "negative100": "#FADBD7",
  "negative1000": "#000",
  "negative200": "#F4AFA7",
  "negative300": "#EB7567",
  "negative400": "#E54937",
  "negative50": "#FDF0EF",
  "negative500": "#AE372A",
  "negative600": "#892C21",
  "negative700": "#5C1D16",
  "negative800": "#471009",
  "negative900": "#1a0603",
  "positive": "#07A35A",
  "positive100": "#CDEDDE",
  "positive1000": "#000",
  "positive200": "#88D3B0",
  "positive300": "#43B982",
  "positive400": "#07A35A",
  "positive50": "#EBF8F2",
  "positive500": "#057C44",
  "positive600": "#046236",
  "positive700": "#034124",
  "positive800": "#000",
  "positive900": "#000",
  "primary": "#276EF1",
  "primary100": "#D2E0FC",
  "primary1000": "#000",
  "primary200": "#9CBCF8",
  "primary300": "#548BF4",
  "primary400": "#276EF1",
  "primary50": "#EDF3FE",
  "primary500": "#174EB6",
  "primary600": "#123D90",
  "primary700": "#0C2960",
  "primary800": "#051c47",
  "primary900": "#020917",
  "primaryFontFamily": "system-ui, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
  "rating": "#FFC043",
  "rating100": "#fff3dc",
  "rating1000": "#100b00",
  "rating200": "#FFE1A5",
  "rating300": "#ffd176",
  "rating400": "#FFC043",
  "rating50": "#fff",
  "rating500": "#ffaf10",
  "rating600": "#dc9200",
  "rating700": "#a97000",
  "rating800": "#764e00",
  "rating900": "#432d00",
  "type": "light",
  "warning": "#ED6F0E",
  "warning100": "#FBE2CF",
  "warning1000": "#000",
  "warning200": "#F6BA8B",
  "warning300": "#F19248",
  "warning400": "#ED6F0E",
  "warning50": "#FEF3EC",
  "warning500": "#B4540B",
  "warning600": "#8E4308",
  "warning700": "#5F2C06",
  "warning800": "#2c1503",
  "warning900": "#000",
  "white": "#fff",
}
`;

exports[`Generate primitives should preserve explicitly defined color varians in a generated primitives object 1`] = `
Object {
  "black": "#000",
  "mono100": "#FFFFFF",
  "mono1000": "#000000",
  "mono200": "#F7F7F7",
  "mono300": "#F0F0F0",
  "mono400": "#E5E5E5",
  "mono50": "#fff",
  "mono500": "#CCCCCC",
  "mono600": "#B3B3B3",
  "mono700": "#999999",
  "mono800": "#666666",
  "mono900": "#333333",
  "negative": "#E54937",
  "negative100": "#FADBD7",
  "negative1000": "#000",
  "negative200": "#F4AFA7",
  "negative300": "#EB7567",
  "negative400": "#E54937",
  "negative50": "#FDF0EF",
  "negative500": "#AE372A",
  "negative600": "#892C21",
  "negative700": "#5C1D16",
  "negative800": "#471009",
  "negative900": "#1a0603",
  "positive": "#07A35A",
  "positive100": "#CDEDDE",
  "positive1000": "#000",
  "positive200": "#88D3B0",
  "positive300": "#43B982",
  "positive400": "#07A35A",
  "positive50": "#EBF8F2",
  "positive500": "#057C44",
  "positive600": "#046236",
  "positive700": "#034124",
  "positive800": "#000",
  "positive900": "#000",
  "primary": "red",
  "primary100": "#f99",
  "primary1000": "#000",
  "primary200": "#f66",
  "primary300": "#f33",
  "primary400": "red",
  "primary50": "#fcc",
  "primary500": "green",
  "primary600": "#900",
  "primary700": "#600",
  "primary800": "#300",
  "primary900": "#000",
  "primaryFontFamily": "system-ui, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
  "rating": "#FFC043",
  "rating100": "#fff3dc",
  "rating1000": "#100b00",
  "rating200": "#FFE1A5",
  "rating300": "#ffd176",
  "rating400": "#FFC043",
  "rating50": "#fff",
  "rating500": "#ffaf10",
  "rating600": "#dc9200",
  "rating700": "#a97000",
  "rating800": "#764e00",
  "rating900": "#432d00",
  "type": "light",
  "warning": "#ED6F0E",
  "warning100": "#FBE2CF",
  "warning1000": "#000",
  "warning200": "#F6BA8B",
  "warning300": "#F19248",
  "warning400": "#ED6F0E",
  "warning50": "#FEF3EC",
  "warning500": "#B4540B",
  "warning600": "#8E4308",
  "warning700": "#5F2C06",
  "warning800": "#2c1503",
  "warning900": "#000",
  "white": "#fff",
}
`;
